base/loading.jl

```diff
@@ -90,10 +90,11 @@ struct SHA1
         return new(bytes)
     end
 end
-SHA1(s::Union{String,SubString{String}}) = SHA1(hex2bytes(s))
+SHA1(s::AbstractString) = SHA1(hex2bytes(s))
 string(hash::SHA1) = bytes2hex(hash.bytes)
+print(io::IO, hash::SHA1) = print(io, string(hash))
```

Comment:
This seems suboptimal:   It seems like we should have a `bytes2hex(io, bytes)` method that writes the hex digits to a stream rather than allocating a string.  That way you can print the hex digits directly instead of allocating a string first.

Reply:
Let's get this merged and tweak for optimality in the future.